python/paddle/fluid/dygraph/static_runner.py

```diff
@@ -0,0 +1,637 @@
+                label = data[1]
+                label.stop_gradient = True
+
+                cost = fc(inputs=img)
+
+                loss = fluid.layers.cross_entropy(cost, label)
+                avg_loss = fluid.layers.mean(loss)
+
+                avg_loss.backward()
+                sgd.minimize(avg_loss)
+    """
+
+    def __init__(self, model_dir, model_filename=None, params_filename=None):
+        super(StaticModelRunner, self).__init__()
+
+        # Step 0. key variable definitions
+        self._load_program_desc = None
+        self._program_desc = None
+        # the layer outputs var desc
+        self._output_descs = []
+        # input, output, params name list
+        self._input_names = []
+        self._output_names = []
+        self._param_names = []
+
+        # Step 1. load program desc from disk
+        # the saved model hold feed, fetch & scale op, no need, can be remove
+        self._load_program_desc = self._load_static_model(model_dir,
+                                                          model_filename)
+
+        # Step 2. set all `is_test` attributes to False
+        self._change_is_test_status(False)
+
+        # Step 3. load all parameters
+        self._load_persisitable_dict(model_dir, params_filename)
+
+        # Step 4. generate backwar program desc
+        self._program_desc = self._append_backward_desc()
+
+        # Step 5. recheck parameters stop gradients
+        self._recheck_stop_gradients()
+
+        # For Debug
+        # DescParser.print_program_desc(self._program_desc)
+
+    def train(self):
+        framework._dygraph_tracer().train_mode()
+        self._change_is_test_status(False)
+
+    def eval(self):
+        framework._dygraph_tracer().eval_mode()
+        self._change_is_test_status(True)
+
+    def forward(self, inputs):
+        # Step 1. check inputs, may affect performance
+        if not isinstance(inputs, (list, tuple)):
+            inputs = [inputs]
+
+        input_vars = []
+        for i, value in enumerate(inputs):
+            if not isinstance(value, (np.ndarray, core.VarBase)):
+                raise TypeError(
+                    "The type of inputs.value in StaticModelRunner.forward must be numpy array or Variable(VarBase), but received %s."
+                    % type(value))
+            # NOTE: In order to unify the API, firstly convert the input to VarBase
+            if isinstance(value, np.ndarray):
+                var = core.VarBase(
+                    value=value,
+                    name=self._input_names[i],
+                    persistable=False,
+                    place=framework._current_expected_place(),
+                    zero_copy=True)
+            else:
+                var = value
+                var.name = self._input_names[i]
+            # NOTE: after PR22939 [Add double grad], the grad op maker's
+            # SetOutput will set to None if the input var stop_gradient=True 
+            var.stop_gradient = False
```

Comment:
Should we reset `stop_gradient` attribute after this layer runs?

Reply:
done, no longer change this attr